### ISSUE TITLE
Flex crash after automatic logout for Delivery calendar

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#261]
_Imported from trac ticket 261.
Created by **kbower**
Opened in trac: Aug 23, 2010, 2:29:45 PM
Last modified in trac: Oct 21, 2010, 6:57:06 PM_
***
_by kbower:_
At least problem on Firefox, maybe all browsers.

Login, bring up an order and open the Delivery calendar chooser.

Allow idle timeout logout.

Log back in, bring up an order and open the Delivery calendar chooser again.  This crash occurs in flex player:
```
TypeError: Error #1009: Cannot access a property or method of a null object reference.
	at mx.controls::DateField/displayDropdown()
	at Function/http://adobe.com/AS3/2006/builtin::apply()
	at mx.core::UIComponent/callLaterDispatcher2()
	at mx.core::UIComponent/callLaterDispatcher()
```
**I believe this is because the logout destroys all modal dialog boxes and this must be being interpreted as one...**

**Comment by kentbower** [comment]
_**kbower** trac comment on Oct 21, 2010, 6:57:06 PM:_

(In [[834]](../commit/f6a3cd801b89cf2b9f97c9c66247b25d25c8766d))
 * Fix destruction of modals during logout (fixes #261)
 * Require authentication for most controllers (see [#356](356)) and fix up client authentication a little

**Comment by kentbower** [comment]
_**kbower** trac status change on Oct 21, 2010, 6:57:06 PM:_ **closed**
